crates/noirc_frontend/src/hir_def/types.rs

```diff
@@ -570,6 +574,7 @@ impl Type {
             | Type::Integer(_, _, _)
             | Type::Bool(_)
             | Type::String(_)
+            | Type::FmtString(_, _)
```

Comment:
We should add cases for String and FmtString here to match the Array case. I think it's an error that they're not already checked.

Reply:
Fixed for both cases